torch/_inductor/codecache.py

```diff
@@ -764,17 +764,21 @@ def _lookup_graph(
 
         # See _save_graph(); we don't store the callable in the cache entry so
         # recreate it here from the PyCodeCache disk cache.
+        artifact_path = get_path(graph.cache_key, "py")[2]
+        if not os.path.exists(artifact_path):
+            counters["inductor"]["fxgraph_lookup_write_file"] += 1
+            write_atomic(artifact_path, graph.source_code)
```

Comment:
One design decision that @masnesral and I discussed here is what if we only write the file to disk when config.debug is enabled and make `PyCodeCache.load_by_key_path` work without files being written to disk. Although @jansel recently changed subproc communication to use files.

Reply:
For the many individual triton kernels we write out it seems unnecessary to write out the file to disk.. For the output code file I think it's worth preserving by default even if debug is not set. Will that still be the case here ?

Reply:
This is only for the output code, not the individual triton kernels. So the behavior on the current PR is always writing the output code. So I guess it is ready for review then.